components/recipeEditIngredientRow.js

```diff
@@ -1,25 +1,43 @@
-import React, { useState } from 'react';
-import { View, Text, TextInput } from 'react-native';
-import styles from '../styles/Recipes/newRecipe';
+import React, {
+  useState,
+  useEffect,
+} from 'react';
+import {
+  View,
+  Text,
+  TextInput,
+} from 'react-native';
+import { Icon } from 'react-native-elements';
+import styles from '../styles/Recipes/formRecipe';
 import formatMoney from '../utils/formatMoney';
 
 function IngredientRow(props) {
-  const { ingredient, totalPrice, setTotalPrice, changeIngredientDataQuantity } = props;
+  const {
+    ingredient,
+    totalPrice,
+    setTotalPrice,
+    changeIngredientDataQuantity,
+    deleteIngredient,
+  } = props;
   const [currentQuantity, setCurrentQuantity] = useState(ingredient.recipeQuantity.toString());
   const priceFactor = ingredient.price / ingredient.quantity;
   const [currentPrice, setCurrentPrice] = useState(priceFactor * ingredient.recipeQuantity);
 
-  function changeQuantity() {
-    if (currentQuantity === '') {
-      setCurrentQuantity('0');
-    }
+  useEffect(() => {
     const lastPrice = currentPrice;
     const newPrice = Number(currentQuantity.replace(',', '.')) * priceFactor;
     ingredient.recipeQuantity = newPrice;
     setCurrentPrice(newPrice);
     setTotalPrice(totalPrice - lastPrice + newPrice);
     ingredient.recipeQuantity = Number(currentQuantity.replace(',', '.'));
     changeIngredientDataQuantity(ingredient.id, Number(currentQuantity.replace(',', '.')));
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currentQuantity]);
```

Comment:
entiendo que la idea de esa regla es que no hayan cosas dentro del efecto que no estén declaradas como dependencias del efecto en el arreglo del final, y en este caso creo que el currentPrice y el priceFactor lo serían o no? Si no me equivoco la idea es evitar que el efecto mismo se "re-evalúe" cada vez que cambien las cosas referenciadas desde adentro, como props, o cosas del state, podi ver más [acá](https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies)